src/middlewares/__tests__/secretsManager.js

```diff
@@ -217,6 +218,56 @@ describe('🔒 SecretsManager Middleware', () => {
     })
   })
 
+  test(`It should fail if "shouldFail" flag provided and call failed`, (done) => {
+    const errorMessage = 'Internal Error / Secret doesn\'t exist'
+    getSecretValueMock.mockReturnValueOnce({
+      promise: () => Promise.reject(new Error(errorMessage))
+    })
+
+    const errHandler = err => {
+      getSecretValueMock.mockClear()
+      expect(err.message).toEqual(errorMessage)
+      done()
+    }
+    return testScenario({
+      mockResponse: {},
+      middlewareOptions: {
+        secrets: {
+          KEY_NAME: 'failed_call'
+        },
+        shouldFail: true
+      },
+      callbacks: [
+        (_, {context}) => {
+          throw new Error('Not supposed to be called')
+        }
+      ],
+      done: errHandler
+    })
+  })
+
+  test(`It should resolve if "shouldFail" flag not provided and call failed`, (done) => {
+    const errorMessage = 'Internal Error / Secret doesn\'t exist'
+    getSecretValueMock.mockReturnValueOnce({
+      promise: () => Promise.reject(new Error(errorMessage))
+    })
+    return testScenario({
+      mockResponse: {},
+      middlewareOptions: {
+        secrets: {
+          KEY_NAME: 'failed_call'
+        }
+      },
+      callbacks: [
+        (_, {context}) => {
```

Comment:
Unused `context` variable here?

Reply:
eslint didn't mention it so I thought that's not an issue but sure, I'll delete it